src/index.js

```diff
@@ -1020,27 +1007,110 @@ export default class AuthnWidget {
         .catch(() => this.generalErrorRenderer(AuthnWidget.COMMUNICATION_ERROR_MSG));
     }
     else {
-      setTimeout(() => {
+      this.pollCheckGetHandler = setTimeout(() => {
         this.pollCheckGet(currentVerificationCode, timeout);
-      }, timeout)
+      }, timeout);
     }
   }
 
-  makeIdVerificationErrorMessage(errorDetails) {
-    var errorMessage = '';
+  async handleRetryVerification(event, cnt) {
+    const maxretries = 10;
+    if (cnt === undefined) cnt = 1;
+    else if (cnt > maxretries) {
+      console.log("exceeded max retries, stop sending retry request");
+      return;
+    }
 
-    errorDetails.forEach(errorDetail =>
-      {
-        if (errorMessage === '') {
-          errorMessage = errorDetail.userMessage
-        }
-        else {
-          var append = '<br>' + errorDetail.userMessage
-          errorMessage += append
-        }
-      })
+    clearTimeout(this.pollCheckGetHandler);
+
+    let actionId = event.target.id;
+    let data = this.store;
+    data.prevState = data.state;
+    data.state = await this.store.reduce('POST_FLOW', actionId);
+    data.state = await this.store.getStore();
+    if (data.prevState.username && !data.state.username) {
+      data.state.username = data.prevState.username;
+    }
+    console.log('dispatching retry: ' + actionId+" #"+cnt);
+    console.log(data.state);
+
+    if (data.state.status === 'ID_VERIFICATION_REQUIRED') {
+      console.log("not returning to first screen, resubmiting post request in 1 second");
+      await new Promise(r => setTimeout(r, 1000));
+      this.handleRetryVerification(event, cnt+1);
+    } else {
+      this.store.notifyListeners();
+    }
+  }
 
-    return errorMessage;
+  handleIdVerificationDevice() {
+    document.getElementById('icon').src = this.store.baseUrl + "/assets/images/newtab.svg";
```

Comment:
This baseUrl might not be PingFederate, it could just be the widget hosted somewhere else. We will need to include this icon in the widget

Reply:
Stored newtab.svg in src/assets/images in the widget, and changed image url to point to the widget-hosted icon. Initially tried to store it at the root level /images folder, but that image url is not available.

Reply:
@khosravian, how should we handle this? We dont really have instructions anywhere describing that you need to copy these images to your hosted app as well.

Maybe encoding this logo into CSS is the easiest path forward?

Reply:
converted svg to css, removed svg file, replaced image with css in device screen